### PR TITLE
fix max concurrent stream config behavior

### PR DESCRIPTION
**Pull request by dlfurse**
introduced single source for default max streams and connected the local max streams variable to the advertised settings in the preface

### CODE REVIEW

File: src/main/java/com/twitter/whiskey/net/SpdyConstants.java [hunk 1]
```diff
@@ -13,6 +13,7 @@
     static final int PRIORITY_LEVELS = 8;
     static final int SPDY_SESSION_STREAM_ID = 0;
     static final int DEFAULT_INITIAL_WINDOW_SIZE = 65536;
+    static final int DEFAULT_MAX_CONCURRENT_STREAMS = 100;
```
Comment (yschimke) [inline src/main/java/com/twitter/whiskey/net/SpdyConstants.java]: shouldn't this default to Integer.MAX_VALUE until the server gives us a value?

File: src/main/java/com/twitter/whiskey/net/SpdySession.java [hunk 2]
```diff
@@ -670,7 +670,7 @@ private void sendData(SpdyStream stream) {
     private void sendClientSettings() {
 
         SpdySettings settings = new SpdySettings();
-        settings.setValue(SpdySettings.MAX_CONCURRENT_STREAMS, 100);
+        settings.setValue(SpdySettings.MAX_CONCURRENT_STREAMS, localMaxConcurrentStreams);
```
Comment (yschimke) [inline src/main/java/com/twitter/whiskey/net/SpdySession.java]: Does the client work for non-push scenarios with the default of 0 in Builder.maxPushStreams?

src/main/java/com/twitter/whiskey/net/SpdySession.java:        localMaxConcurrentStreams = configuration.getMaxPushStreams();

File: src/main/java/com/twitter/whiskey/net/SpdyConstants.java (hunk 1 above)
Comment (dlfurse) [inline src/main/java/com/twitter/whiskey/net/SpdyConstants.java]: yes, definitely - SPDY 3.1 spec says there is no maximum by default so we can just saturate the data type.

File: src/main/java/com/twitter/whiskey/net/SpdySession.java (hunk 2 above)
Comment (dlfurse) [inline src/main/java/com/twitter/whiskey/net/SpdySession.java]: yes, just verified that a simple get with no pushes works if the default is zero.

**Comment by madster26** [comment]
LGTM

**Comment by kgoodier** [comment]
Thanks Daniel!

**Pull request merged**